src/main/java/cceclipseplugin/editor/listeners/DirectoryListener.java

```diff
@@ -0,0 +1,203 @@
+package cceclipseplugin.editor.listeners;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+
+import cceclipseplugin.core.EclipseRequestManager;
+import cceclipseplugin.core.PluginManager;
+import dataMgmt.MetadataManager;
+import dataMgmt.models.FileMetadata;
+import dataMgmt.models.ProjectMetadata;
+import requestMgmt.RequestManager;
+import websocket.models.notifications.FileCreateNotification;
+import websocket.models.notifications.FileDeleteNotification;
+import websocket.models.notifications.FileMoveNotification;
+import websocket.models.notifications.FileRenameNotification;
+import websocket.models.responses.FileCreateResponse;
+
+public class DirectoryListener extends AbstractDirectoryListener {
+	
+	/**
+	 * Handles a resource delta in the case that the resource is and IProject.
+	 * 
+	 * @param delta
+	 * @return 
+	 * 		true if the recursion down the delta's children stops after this handler
+	 */
+	@Override
+	protected boolean handleProject(IResourceDelta delta) {
+		IProject p = (IProject) delta.getResource();
+		ProjectMetadata projectMeta = PluginManager.getInstance().getMetadataManager().getProjectMetadata(p.getFullPath().toString());
+		RequestManager rm = PluginManager.getInstance().getRequestManager();
+		
+		if (delta.getKind() == IResourceDelta.REMOVED) {
+			System.out.println("flag: " + delta.getFlags());
```

Comment:
I don't think we should be printing these out every time, maybe below is fine though